### PR TITLE
[ALLUXIO-2426] Try using authenticated and login user rules to deflake PermissionCheckTest

### PR DESCRIPTION
**Pull request by apc999**
https://alluxio.atlassian.net/browse/ALLUXIO-2426

`PermissionCheckTest ` still fails but for `verifySetAcl` now. This PR continues cleaning up the code by replacing setting the singleton by using rules for authenticated and login user.

### CODE REVIEW

File: core/common/src/test/java/alluxio/security/LoginUserTestUtils.java [hunk 1]
```diff
@@ -42,9 +39,7 @@ public static void resetLoginUser() {
    */
   public static void resetLoginUser(String user) throws IOException {
     synchronized (LoginUser.class) {
-      resetLoginUser();
-      Configuration.set(PropertyKey.SECURITY_LOGIN_USERNAME, user);
```
Comment (apc999) [inline core/common/src/test/java/alluxio/security/LoginUserTestUtils.java]: this modifies the configuration singleton, not ideal.

**Comment by apc999** [comment]
Jenkins, test this please.

**Comment by jsimsa** [comment]
Jenkins, test this please

File: core/common/src/test/java/alluxio/SetAndRestoreLoginUser.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import alluxio.security.User;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * An AutoCloseable which temporarily modifies process login user when it is
+ * constructed and restores the previous login user when it is closed.
+ */
+@ThreadSafe
+public final class SetAndRestoreLoginUser implements AutoCloseable {
```
Comment (jsimsa) [inline core/common/src/test/java/alluxio/SetAndRestoreLoginUser.java]: would it make sense to make this a nested class of LoginUserRule?

File: core/common/src/test/java/alluxio/SetAndRestoreAuthenticatedUser.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import alluxio.security.authentication.AuthenticatedClientUser;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * An AutoCloseable which temporarily modifies authenticated user on server side when it is
+ * constructed and restores the authenticated user when it is closed.
+ */
+@ThreadSafe
+public final class SetAndRestoreAuthenticatedUser implements AutoCloseable {
```
Comment (jsimsa) [inline core/common/src/test/java/alluxio/SetAndRestoreAuthenticatedUser.java]: would it make sense to make this a nested class of AuthenticatedUserRule?

**Comment by jsimsa** [comment]
@apc999 left a couple of comments ... the test failure looks like a deadlock to me ... we should add a conservative timeout to each test (e.g. 120 seconds) to make sure we can identify the tests that hang

**Comment by apc999** [comment]
Jenkins, test this please

**Comment by apc999** [comment]
Jenkins, test this please

File: core/common/src/test/java/alluxio/SetAndRestoreAuthenticatedUser.java (hunk 3 above)
Comment (apc999) [inline core/common/src/test/java/alluxio/SetAndRestoreAuthenticatedUser.java]: @jsimsa  I will create a separate PR to consolidate these rules, leaving this PR small

File: core/common/src/test/java/alluxio/SetAndRestoreLoginUser.java (hunk 2 above)
Comment (apc999) [inline core/common/src/test/java/alluxio/SetAndRestoreLoginUser.java]: @jsimsa as mentioned above. Let's see the PR

**Comment by apc999** [comment]
@jsimsa @ifcharming  after I merged this PR with master, no failure is seen. I suspect the previous build issue is not introduced by this PR. 

PTAL

File: core/common/src/test/java/alluxio/SetAndRestoreLoginUser.java (hunk 2 above)
Comment (apc999) [inline core/common/src/test/java/alluxio/SetAndRestoreLoginUser.java]: @jsimsa PTAL https://github.com/apc999/alluxio/pull/1/files

**Comment by apc999** [comment]
Jenkins, test this please

**Comment by apc999** [comment]
Jenkins, test this please

File: core/common/src/test/java/alluxio/SetAndRestoreAuthenticatedUser.java (hunk 3 above)
Comment (jsimsa) [inline core/common/src/test/java/alluxio/SetAndRestoreAuthenticatedUser.java]: ok

**Comment by jsimsa** [comment]
@ifcharming PTAL

**Comment by ifcharming** [comment]
@apc999 mostly LGTM, do we also need to update integration tests which set `LoginUser` and `AuthenticatedUser` ?

**Comment by apc999** [comment]
@ifcharming there is a follow up PR to further refactor and clean the tests
https://github.com/apc999/alluxio/pull/1/files
After this PR merged, I will re-direct that PR.

**Comment by ifcharming** [comment]
LGTM, thanks for the fix!

**Pull request merged**